test/terraform_aws_nextjs_serverless_test.go

```diff
@@ -0,0 +1,166 @@
+package test
+
+import (
+	"crypto/tls"
+	"fmt"
+	"os"
+	"strings"
+	"testing"
+
+	http_helper "github.com/gruntwork-io/terratest/modules/http-helper"
+	"github.com/gruntwork-io/terratest/modules/terraform"
+	test_structure "github.com/gruntwork-io/terratest/modules/test-structure"
+	"github.com/stretchr/testify/assert"
+	"github.com/tidwall/gjson"
+)
+
+func TestTerraformCloudFrontServerlessNextJS(t *testing.T) {
+	t.Parallel()
+
+	terraformDir := "../examples/nextjs-v13/terraform/"
+	varFiles := []string{"../../../config/terratest.tfvars"}
+	fmt.Println(os.Getwd())
+
+	// Normally, a terraform options loading would be cleaner here,
+	// but there is a bug in the library implementation that resets the saved location of the terraform output file
+	// Thus it's safer to use terraform base dir and var file variables throughout this test.
+
+	// This is the cleanup part
+	defer test_structure.RunTestStage(t, "cleanupNextJSCloudFront", func() {
+		undeployTerraform(t, terraformDir, varFiles)
+	})
+
+	// This is the setup
+	test_structure.RunTestStage(t, "initializeNextJSCloudFront", func() {
+		deployTerraform(t, terraformDir, varFiles)
+	})
+
+	// To get the URL, we need to parse the terraform output
+	terraformOptions := terraform.WithDefaultRetryableErrors(t, &terraform.Options{
+		TerraformDir: terraformDir,
+		Reconfigure:  true,
+		VarFiles:     varFiles,
+	})
+	serverlessModule := terraform.OutputJson(t, terraformOptions, "next_serverless")
+	resultStat := gjson.Parse(fmt.Sprint(serverlessModule))
+	aliasURL := resultStat.Get("static-deploy.next_distribution.aliases.0").String()
+	if aliasURL == "" {
+		aliasURL = resultStat.Get("static-deploy.next_distribution.domain_name").String()
+	}
+
+	// Stage 1
+	test_structure.RunTestStage(t, "serverHealthCheck", func() {
+		hitTheFrontPage(t, aliasURL)
+	})
+
+	// Stage 2
+	test_structure.RunTestStage(t, "publicAssetsCheck", func() {
+		getIcons(t, aliasURL)
+	})
+
+	// Stage 3
+	// I could've checked the S3 buckets to see if these resources are uploaded there, too
```

Comment:
pls remove this comment, not necessary for our open source module

Reply:
Removed